ethcore/src/engines/clique/mod.rs

```diff
@@ -198,13 +198,15 @@ impl Clique {
 			step_service: None,
 		};
 
-		let res = Arc::new(engine);
+		let mut engine= Arc::new(engine);
 
 		if our_params.period > 0 {
-			engine.step_service = Some(StepService::start(Arc::downgrade(&res) as Weak<Engine<_>>));
+			let engine_weak = Arc::downgrade(&engine.clone());
```

Comment:
`clone()` doesn't seem to be needed here
```suggestion
			let engine_weak = Arc::downgrade(&engine);
```